pandas/formats/format.py

```diff
@@ -1951,7 +1951,11 @@ def _format_hierarchical_rows(self):
                 for spans, levels, labels in zip(level_lengths,
                                                  self.df.index.levels,
                                                  self.df.index.labels):
-                    values = levels.take(labels)
+
+                    values = levels.take(labels,
+                                         allow_fill=levels._can_hold_na,
```

Comment:
@jorisvandenbossche So you claim that if I substitute

    values = levels.take(labels,
                         allow_fill=levels._can_hold_na,
                         fill_value=True)

with

    values = levels.take(labels,
                         fill_value=True)

that no test will detect the change, i.e. that all tests pass? I ran the tests by entering

    $ nosetests pandas/io/tests/test_excel.py:Openpyxl20Tests

against the PR which contains `allow_fill=levels._can_hold_na` and then applied the aforementioned substitution and ran the tests again. In the first case the output is

    ......SS...........................
    ----------------------------------------------------------------------
    Ran 35 tests in 2.322s

    OK (SKIP=2)

and in the second case it is

    ......SS...............EEE.........
    ======================================================================
    ERROR: test_to_excel_multiindex (pandas.io.tests.test_excel.Openpyxl20Tests)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/tests/test_excel.py", line 1776, in wrapped
        orig_method(self, *args, **kwargs)
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/tests/test_excel.py", line 1321, in test_to_excel_multiindex
        frame.to_excel(path, 'test1', header=False)
      File "/home/mpuels/progs/pandas-mpuels/pandas/core/frame.py", line 1431, in to_excel
        startrow=startrow, startcol=startcol)
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/excel.py", line 875, in write_cells
        for cell in cells:
      File "/home/mpuels/progs/pandas-mpuels/pandas/formats/format.py", line 1986, in get_formatted_cells
        self._format_body()):
      File "/home/mpuels/progs/pandas-mpuels/pandas/formats/format.py", line 1957, in _format_hierarchical_rows
        fill_value=True)
      File "/home/mpuels/progs/pandas-mpuels/pandas/indexes/base.py", line 1438, in take
        raise ValueError(msg.format(self.__class__.__name__))
    ValueError: Unable to fill values because Int64Index cannot contain NA
    
    ======================================================================
    ERROR: test_to_excel_multiindex_cols (pandas.io.tests.test_excel.Openpyxl20Tests)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/tests/test_excel.py", line 1776, in wrapped
        orig_method(self, *args, **kwargs)
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/tests/test_excel.py", line 1366, in test_to_excel_multiindex_cols
        frame.to_excel(path, 'test1', merge_cells=self.merge_cells)
      File "/home/mpuels/progs/pandas-mpuels/pandas/core/frame.py", line 1431, in to_excel
        startrow=startrow, startcol=startcol)
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/excel.py", line 875, in write_cells
        for cell in cells:
      File "/home/mpuels/progs/pandas-mpuels/pandas/formats/format.py", line 1986, in get_formatted_cells
        self._format_body()):
      File "/home/mpuels/progs/pandas-mpuels/pandas/formats/format.py", line 1957, in _format_hierarchical_rows
        fill_value=True)
      File "/home/mpuels/progs/pandas-mpuels/pandas/indexes/base.py", line 1438, in take
        raise ValueError(msg.format(self.__class__.__name__))
    ValueError: Unable to fill values because Int64Index cannot contain NA
    
    ======================================================================
    ERROR: test_to_excel_multiindex_dates (pandas.io.tests.test_excel.Openpyxl20Tests)
    ----------------------------------------------------------------------
    Traceback (most recent call last):
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/tests/test_excel.py", line 1776, in wrapped
        orig_method(self, *args, **kwargs)
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/tests/test_excel.py", line 1387, in test_to_excel_multiindex_dates
        tsframe.to_excel(path, 'test1', merge_cells=self.merge_cells)
      File "/home/mpuels/progs/pandas-mpuels/pandas/core/frame.py", line 1431, in to_excel
        startrow=startrow, startcol=startcol)
      File "/home/mpuels/progs/pandas-mpuels/pandas/io/excel.py", line 875, in write_cells
        for cell in cells:
      File "/home/mpuels/progs/pandas-mpuels/pandas/formats/format.py", line 1986, in get_formatted_cells
        self._format_body()):
      File "/home/mpuels/progs/pandas-mpuels/pandas/formats/format.py", line 1957, in _format_hierarchical_rows
        fill_value=True)
      File "/home/mpuels/progs/pandas-mpuels/pandas/indexes/base.py", line 1438, in take
        raise ValueError(msg.format(self.__class__.__name__))
    ValueError: Unable to fill values because Int64Index cannot contain NA
    
    ----------------------------------------------------------------------
    Ran 35 tests in 2.163s
    
    FAILED (SKIP=2, errors=3)

So the test which I constructed for this PR did not fail, but three other tests which already exist. Shall I nonetheless construct another test case where the Index contains integers? Or did I miss anything? Thanks for your patience!